src/moonlink_backend/src/error.rs

```diff
@@ -1,89 +1,211 @@
 use moonlink::Error as MoonlinkError;
 use moonlink_connectors::Error as MoonlinkConnectorError;
 use moonlink_connectors::PostgresSourceError;
+use moonlink_error::{ErrorStatus, ErrorStruct};
 use moonlink_metadata_store::error::Error as MoonlinkMetadataStoreError;
 use std::num::ParseIntError;
+use std::panic::Location;
 use std::result;
 use std::sync::Arc;
 use thiserror::Error;
 
 /// Custom error type for moonlink_backend
 #[derive(Clone, Debug, Error)]
 pub enum Error {
-    #[error("Parse integer error: {source}")]
-    ParseIntError { source: ParseIntError },
-    #[error("Postgres source error: {source}")]
-    PostgresSource { source: Arc<PostgresSourceError> },
-    #[error("IO error: {source}")]
-    Io { source: Arc<std::io::Error> },
-    #[error("Moonlink connector error: {source}")]
-    MoonlinkConnectorError { source: MoonlinkConnectorError },
-    #[error("Moonlink error: {source}")]
-    MoonlinkError { source: MoonlinkError },
-    #[error("Moonlink metadata error: {source}")]
-    MoonlinkMetadataStoreError { source: MoonlinkMetadataStoreError },
+    #[error("{0}")]
+    ParseIntError(ErrorStruct),
+
+    #[error("{0}")]
+    PostgresSource(ErrorStruct),
+
+    #[error("{0}")]
+    Io(ErrorStruct),
+
+    #[error("{0}")]
+    MoonlinkConnectorError(ErrorStruct),
+
+    #[error("{0}")]
+    MoonlinkError(ErrorStruct),
+
+    #[error("{0}")]
+    MoonlinkMetadataStoreError(ErrorStruct),
+
     #[error("Invalid argument: {0}")]
     InvalidArgumentError(String),
```

Comment:
No need in this PR but it need to adapt all error types at the end of the day; one big motivation is source error location